.github/workflows/transport.yaml

```diff
@@ -0,0 +1,212 @@
+        # Only check out the following files to limit bandwidth usage.
+        # Respectively
+        # 1. IEA Extended World Energy Balances, OECD source, 2022 edition
+        sparse-checkout: |
+          iea/372f7e29-en.zip
+        sparse-checkout-cone-mode: false
+
+    - name: Connect static data to message_data
+      run: cp -rsv $(realpath message-static-data)/* message_data/data/
+
+    - uses: actions/checkout@v4
+      with:
+        repository: iiasa/MESSAGE_Buildings
+        path: buildings
+        ssh-key: ${{ secrets.MESSAGE_BUILDINGS_PRIVATE_KEY }}
+
+    # NB Needed for self-hosted runner, not for GitHub
+    - uses: actions/setup-java@v4
+      with:
+        distribution: temurin
+        java-version: "17"
+
+    - uses: actions/setup-python@v5
+      with:
+        # Latest version testable on GitHub Actions
+        python-version: "3.12"
+        cache: pip
+        cache-dependency-path: "**/pyproject.toml"
+
+    - uses: iiasa/actions/setup-gams@main
+      with:
+        version: 35.2.0
+        license: ${{ secrets.GAMS_LICENSE }}
+
+    # NB the IIASA-hosted runner does not have permissions to do "sudo apt-get",
+    #    so the following doesn't work. Rather, the necessary packages
+    #    (graphviz libgraphviz-dev pkg-config) are manually installed on the
+    #    host system
+    # - uses: ts-graphviz/setup-graphviz@v2
+
+    # By default, the next step installs dependencies from PyPI.
+    # To run against unreleased code (on `main`, or other branches for open
+    # PRs), uncomment or modify the lines below as appropriate.
+    # - genno and message-ix-models are released frequently and *should* be
+    #   installed from PyPI, with only temporary exceptions.
+    # - ixmp and message-ix are released less frequently and *may* be installed
+    #   from `main`
+    - name: Force upgrade to unreleased dependencies
+      run: |
+        # pip install --upgrade "genno @ git+https://github.com/khaeru/genno.git@main"
+        pip install --upgrade "ixmp @ git+https://github.com/iiasa/ixmp.git@main"
+        pip install --upgrade "message_ix @ git+https://github.com/iiasa/message_ix.git@main"
+        # Temporary, pending a release that includes iiasa/message-ix-models#161, #166
+        pip install --upgrade "message-ix-models[iea-web] @ git+https://github.com/iiasa/message-ix-models.git@main"
+
+        # Upgrade matplotlib
+        # pyam pins matplotlib < 3.7.1, which conflicts with plotnine per
+        # https://github.com/has2k1/plotnine/pull/751. Override.
+        pip install --upgrade matplotlib
+
+    - name: Install package; upgrade dependencies
+      working-directory: message_data
+      run: pip install --upgrade --upgrade-strategy=eager --editable .[dl,scgen,tests,transport]
+
+    - name: Configure
+      run: |
+        from pathlib import Path
+
+        # Ensure keys are defined
+        import message_data.model.buildings
+        from ixmp import config
+
+        config.add_platform(
```

Comment:
Shouldn't the connection details of `ixmp_dev` be a Secret?

Reply:
Yes, I can convert these to such.

Reply:
I pushed a commit with this change.

To be clear, the current practice of all users using the same, guess-able username and password for these databases does not provide any security. Security is rather provided by limiting access to those on the IIASA network, and then limiting who has access to this network.

If/when these database servers are reconfigured to allow connections from outside the IIASA network—but ideally sooner—individual users should be issued individual usernames and passwords, or tokens, allowing to control access in a more fine-grained way, and the single/shared username-password pair must be permanently revoked. We could then generate specific tokens for use in CI in each of our repositories that need access to the database(s).